packages/components/src/custom-gradient-bar/control-points.js

```diff
@@ -73,23 +73,17 @@ function ControlPointButton( { isOpen, position, color, ...additionalProps } ) {
 	);
 }
 
-function GradientColorPickerDropdown( {
-	isRenderedInSidebar,
-	gradientPickerDomRef,
-	...props
-} ) {
+function GradientColorPickerDropdown( { isRenderedInSidebar, ...props } ) {
 	const popoverProps = useMemo( () => {
 		const result = {
 			className:
 				'components-custom-gradient-picker__color-picker-popover',
-			position: 'top',
 		};
-		if ( isRenderedInSidebar ) {
-			result.anchorRef = gradientPickerDomRef.current;
-			result.position = 'bottom left';
+		if ( ! isRenderedInSidebar ) {
+			result.placement = 'top';
```

Comment:
I realise it's a bit like this in trunk, but this line makes the duotone color picker look like this, overlapping the block toolbar:
![Screen Shot 2022-08-09 at 12 50 18 pm](https://user-images.githubusercontent.com/677833/183566956-dd99f1d9-cb8e-4b01-abfa-29b0477f199a.png)

It might be good to simplify it and have `GradientColorPickerDropdown` always positioned underneath the control point. When there's not enough space `Popover` will already flip it to the `top`.

Would be good to get more thoughts on this from @jasmussen and possibly @ajlende.

Another note is that the `useMemo` here could be removed, as `CustomColorPickerDropdown` doesn't respect it and creates a new object on every render.

Reply:
Just pushed https://github.com/WordPress/gutenberg/pull/42989/commits/1d30a2a52808ba02ce90780cde8132419dec7c40 in which:

- the `placement` is always set to `bottom` regardless of the `isInSidebar` prop (I also tweaked the `offset` by a couple of px)
- I've used `useMemo` also inside `CustomColorPickerDropdown` (instead of removing it)

Reply:
Actually, after https://github.com/WordPress/gutenberg/pull/42989/commits/7c2352845d08edfac00d31e0b47f16a049f03e80, there's currently no need for any `popoverProps` override (although there will probably be a need for it again soon)